tools/topology/topology2/get_abi.sh

```diff
@@ -11,6 +11,29 @@ cat <<EOF_HEADER
 Object.Base.manifest."sof_manifest" {
 	Object.Base.data."SOF ABI" {
 EOF_HEADER
-printf '\t\tbytes\t"0x%02x,' "$ABI_MAJOR"
-printf "0x%02x," "$ABI_MINOR"
-printf '0x%02x\"\n\t}\n}' "$ABI_PATCH"
+
+print_1byte()
+{
+	printf '0x%02x' "$(("$1" & 0xff))"
+}
+
+print_2bytes()
+{
+	printf '0x%02x,0x%02x' "$(("$1" & 0xff))" "$(($(("$1" & 0xff00)) >> 8))"
```

Comment:
```suggestion
	printf '0x%02x,0x%02x' "$(("$1" & 0xff))" "$((("$1" & 0xff00) >> 8))"
```

Reply:
> > the point is that I need to know if I am building a topology for IPC4 or IPC3 before I execute this script and it could be different for each file. I could move the execute process inside the cavs/CMakeLists.txt if you want to avoid 2 scripts. ok i moved it to the cavs CMakeLists.txt now. better?
> 
> I'm afraid I don't really understand the problem and why you moved it one level down to cavs/.
> 
> `execute_process()` runs only one globally and only at configure time, `execute_process()` does not run at build time. So you cannot have a different abi.conf for each topology right now, that's not possible.
> 
> Would two files abi-ipc4.conf and abi-ipc3.conf help?

fixed the suggested change. As for having 2 separate abi files, Im not sure whats simpler. In a way, having each individual folder generate its own ABI is more straightforward I think. It is highly unlikely that we will mix up ipc3 topologies and ipc4 topologies in the same folder.